pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -261,9 +289,26 @@ func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.S
 	if len(serverList) == 0 {
 		return nil, ErrNotFound
 	} else if len(serverList) > 1 {
-		return nil, ErrMultipleResults
+		var lastMatch *servers.Server
+		// try to find exactMatch
+		for _, server := range serverList {
+			if server.Name == exactMatch {
```

Comment:
I checked that if there are two servers with the same Name, kubelet fails to start on them:

```
E1109 15:01:13.517363    9576 kubelet.go:929] Unable to construct api.Node object for kubelet: failed to get external ID from cloud provider: Multiple results where only one expected
```

Do we need to care about these use cases?